test/api/python/sep_log_api.py

```diff
@@ -0,0 +1,240 @@
+##############################################################################
+# Top contributors (to current version):
+#   Yoni Zohar
+#
+# This file is part of the cvc5 project.
+#
+# Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+# in the top-level source directory and their institutional affiliations.
+# All rights reserved.  See the file COPYING in the top-level source
+# directory for licensing information.
+# ############################################################################
+#
+# Two tests to validate the use of the separation logic API.
+#
+# First test validates that we cannot use the API if not using separation
+# logic.
+#
+# Second test validates that the expressions returned from the API are
+# correct and can be interrogated.
+##
+
+import pycvc5
+from pycvc5 import kinds
+
+
+# Test function to validate that we *cannot* obtain the heap/nil expressions
+# when *not* using the separation logic theory
+def validate_exception():
+    slv = pycvc5.Solver()
+    # Setup some options for cvc5 -- we explictly want to use a simplistic
+    # theory (e.g., QF_IDL)
+    slv.setLogic("QF_IDL")
+    slv.setOption("produce-models", "true")
+    slv.setOption("incremental", "false")
+
+    # Our integer type
+    integer = slv.getIntegerSort()
+
+    # we intentionally do not set the separation logic heap
+
+    # Our SMT constants
+    x = slv.mkConst(integer, "x")
+    y = slv.mkConst(integer, "y")
+
+    # y > x
+    y_gt_x = slv.mkTerm(kinds.Gt, y, x)
+
+    # assert it
+    slv.assertFormula(y_gt_x)
+
+    # check
+    r = slv.checkSat()
+
+    # If this is UNSAT, we have an issue so bail-out
+    if not r.isSat():
+
+        return -1
+
+    # We now try to obtain our separation logic expressions from the solver --
+    # we want to validate that we get our expected exceptions.
+
+    caught_on_heap = False
+    caught_on_nil = False
+
+    # The exception message we expect to obtain
+    expected = \
+        "Cannot obtain separation logic expressions if not using the separation " \
+        "logic theory."
+
+    # test the heap expression
+    try:
+        heap_expr = slv.getValueSepHeap()
+    except RuntimeError as e:
+        caught_on_heap = True
+        # Check we get the correct exception string
+        if str(e) != expected:
+            return -1
```

Comment:
Use boolean return values for this function and return true when everything behaves as expected.